src/query/frontend/opencypher/grammar/Cypher.g4

```diff
@@ -191,9 +191,12 @@ relationshipTypes : ':' relTypeName ( '|' ':'? relTypeName )* ;
 
 nodeLabels : nodeLabel ( nodeLabel )* ;
 
-nodeLabel : ':' (labelName | expression2b);
+nodeLabel : ':' labelName ;
 
-labelName : symbolicName | parameter;
+labelName : symbolicName
+          | parameter
+          | variable ( propertyLookup )*
+          ;
```

Comment:
Why was this changed?
I also saw some slack thread where you talked about the possibility of using the parameter instead of just a variable.

Reply:
`Epxression2b` is changed into a `variable`, there is no particular reason why I moved it after `labelName`, instead of `nodeLabel`, it only looks more consistent and it's easier to read since the `nodeLabel` expands into `labelName` and `labelName` into `symbolicName` and `parameter`. The problem with a `variable` is that it expands into a `symbolicName`. `symbolicName` is used to declare a label on its own, so it's not possible to distinguish between a `variable` and a `symbolicName` (it's possible in semantic analysis, but it's not the right way to do it). Example:
`WITH "something" as label CREATE (:label {id: label});`
`Variable (propertyLookup)*` is unambiguous because of the `.`, that is why the grammar looks this way. 
NOTE: Merging this PR we won't be able to use a `variable` to create a label. We'll be able to do it by the value of a property.

Reply:
Even though it won't make a change in how the Memgraph behaves, I'll change `variable ( propertyLookup )* `into `variable ( propertyLookup )+` , so it is easier to understand what happens here.